integreat_cms/cms/utils/content_translation_utils.py

```diff
@@ -0,0 +1,170 @@
+                link.classes.remove("link-external")
+                logger.debug("Removed class 'link-external' from %r", tostring(link))
+
+    # Update internal links
+    for link in content.iter("a"):
+        if translation := internal_link_utils.update_link(link, language_slug):
+            new_url, new_html = translation
+            if new_url != unquote(link.get("href", "")):
+                link.set("href", new_url)
+                logger.debug(
+                    "Updated link url from %s to %s", link.get("href"), new_url
+                )
+
+            if new_html is not None:
+                logger.debug("Updated link text from %s to %s", link.text, new_html)
+                for child in link:
+                    link.remove(child)
+                if isinstance(new_html, str):
+                    link.text = unescape(new_html)
+                else:
+                    link.text = ""
+                    link.append(new_html)
+
+    # Scan for media files in content and replace alt texts
+    for image in content.iter("img"):
+        logger.debug("Image tag found in content (src: %s)", image.attrib["src"])
+        # Remove host
+        relative_url = urlparse(image.attrib["src"]).path
+        # Remove media url prefix if exists
+        if relative_url.startswith(settings.MEDIA_URL):
+            relative_url = relative_url[len(settings.MEDIA_URL) :]
+        # Check whether media file exists in database
+        media_file = MediaFile.objects.filter(
+            Q(file=relative_url) | Q(thumbnail=relative_url)
+        ).first()
+        # Replace alternative text
+        if media_file and media_file.alt_text:
+            logger.debug("Image alt text replaced: %r", media_file.alt_text)
+            image.attrib["alt"] = media_file.alt_text
+
+    return tostring(content, with_tail=False).decode("utf-8")
+
+
+def update_links_to(
+    content_translation: AbstractContentTranslation, user: User | None
+) -> None:
+    """
+    Updates all content translations with links that point to the given translation.
+    This fixes potentially outdated links.
+
+    :param content_translation: The content translation for which links that points to it should get updated
+    :param user: The user who should be responsible for updating the links
+    """
+    for outdated_content_translation in get_referencing_translations(
+        content_translation
+    ):
+        # Assert that the related translation is not archived
+        # Note that this should not be possible, since links to archived pages get deleted
+        if getattr(outdated_content_translation.foreign_object, "archived", False):
+            continue
+
+        new_content = get_cleaned_content(
+            outdated_content_translation.content,
+            outdated_content_translation.language.slug,
+        )
+        if new_content == outdated_content_translation.content:
+            continue
```

Comment:
What's this check for?

Reply:
This is just an optimization so that no new translations get created when the content did not change. For example, this function gets executed when the status of a content translation changes to `PUBLIC`, but its title and content might not have changed and all existing links are still valid. In this case there would be no point in saving new versions with the same content.